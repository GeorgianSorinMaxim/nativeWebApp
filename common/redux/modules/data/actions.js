// @flow

export const APP_LAUNCHED = 'data/APP_LAUNCHED';

export const START_FETCH_DATA = 'data/START_FETCH';

export const DATA_FETCH_SUCCESS = 'data/FETCH_SUCCESS';
export const DATA_FETCH_FAILURE = 'data/FETCH_FAILURE';

type AppLaunched = {
  type: 'data/APP_LAUNCHED'
};

type StartDataFetch = {
  type: 'data/START_FETCH'
};

type DataFetchSuccess = {
  type: 'data/FETCH_SUCCESS',
  payload: Object
};

type DataFetchFailure = {
  type: 'data/FETCH_FAILURE',
  payload: Object
};

export function appLaunched(): AppLaunched {
  return { type: APP_LAUNCHED };
}

export function startFetchingData(): StartDataFetch {
  return { type: START_FETCH_DATA };
}

export function dataFetchSuccess(payload: Object): DataFetchSuccess {
  return { type: DATA_FETCH_SUCCESS, payload };
}

export function dataFetchFailure(error: Object): DataFetchFailure {
  return { type: DATA_FETCH_FAILURE, payload: { error } };
}

export type Action =
  | AppLaunched
  | StartDataFetch
  | DataFetchSuccess
  | DataFetchFailure;
