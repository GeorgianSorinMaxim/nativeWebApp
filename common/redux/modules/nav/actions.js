// @flow
import { NavigationActions } from 'react-navigation';

const navToPath = path => NavigationActions.navigate({ routeName: path });
// const navToPathWithParams = (path: string, params: Object) =>
//   NavigationActions.navigate({ routeName: path, params });

const navToHome = () => navToPath('Home');
const navToFeature = () => navToPath('Feature');

export const actions = {
  navToHome,
  navToFeature
};
