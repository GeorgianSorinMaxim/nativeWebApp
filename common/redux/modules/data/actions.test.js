import * as actions from './actions';

describe('Actions', () => {
  describe('appLaunched', () => {
    it('should return an app launched action', () => {
      const action = actions.appLaunched();
      expect(action).toMatchSnapshot();
    });
  });

  describe('startFetchingData', () => {
    it('should return a start fetching data action', () => {
      const action = actions.startFetchingData();
      expect(action).toMatchSnapshot();
    });
  });

  describe('dataFetchSuccess', () => {
    it('should return a data fetch success action', () => {
      const action = actions.dataFetchSuccess();
      expect(action).toMatchSnapshot();
    });
  });

  describe('dataFetchFailure', () => {
    it('should return a data fetch failure action', () => {
      const action = actions.dataFetchFailure();
      expect(action).toMatchSnapshot();
    });
  });
});
