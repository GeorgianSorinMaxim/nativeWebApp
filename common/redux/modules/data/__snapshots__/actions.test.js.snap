// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Actions appLaunched should return an app launched action 1`] = `
Object {
  "type": "data/APP_LAUNCHED",
}
`;

exports[`Actions dataFetchFailure should return a data fetch failure action 1`] = `
Object {
  "payload": Object {
    "error": undefined,
  },
  "type": "data/FETCH_FAILURE",
}
`;

exports[`Actions dataFetchSuccess should return a data fetch success action 1`] = `
Object {
  "payload": undefined,
  "type": "data/FETCH_SUCCESS",
}
`;

exports[`Actions startFetchingData should return a start fetching data action 1`] = `
Object {
  "type": "data/START_FETCH",
}
`;
